src/lib/operators/join_index.cpp

```diff
@@ -202,40 +208,85 @@ std::shared_ptr<const Table> JoinIndex::_on_execute() {
       }
     }
   } else {  // DATA JOIN since only inner joins are supported for a reference table on the index side
-    // Scan all chunks for index input
-    const auto chunk_count_index_input_table = _index_input_table->chunk_count();
-    for (ChunkID index_chunk_id{0}; index_chunk_id < chunk_count_index_input_table; ++index_chunk_id) {
-      const auto index_chunk = _index_input_table->get_chunk(index_chunk_id);
-      Assert(index_chunk, "Physically deleted chunk should not reach this point, see get_chunk / #1686.");
+    // Here we prefer to use table indexes if the join supports them. If no table index exists or other predicates than
+    // Equals or NotEquals are requested, chunk indexes are used. If no chunk index exists, NestedLoopJoin is used as a
+    // fallback solution.
+    const auto& table_indexes = _index_input_table->get_table_indexes(_index_column_id);
+
+    // table-based index join
```

Comment:
I think this line does not provide additional information that is not communicated with the actual code.